### PR TITLE
return 400 instead of 500 in transport.go

### PR DESCRIPTION
**Pull request by mtrinh11**
Changed addwebhookrequestdecoder to return 400 instead of 500 for bad request bodies and validation failures

### CODE REVIEW

**Review by kristinapathak — APPROVED** [review]

**Comment by mtrinh11** [comment]
fixes https://github.com/xmidt-org/ancla/issues/14

File: transport.go [hunk 1]
```diff
@@ -78,7 +78,7 @@ func addWebhookRequestDecoder(config transportConfig) kithttp.DecodeRequestFunc
 	return func(c context.Context, r *http.Request) (request interface{}, err error) {
 		requestPayload, err := ioutil.ReadAll(r.Body)
 		if err != nil {
-			return nil, err
+			return nil, &erraux.Error{Err: err, Message: "bad request body", Code: http.StatusBadRequest}
```
Comment (joe94) [inline transport.go:81]: I could be wrong but I think this could also happen due to network errors which is not a bad user request

Comment (joe94) [reply transport.go:81]: I had a hard time finding more specific documentation but here's one SOF question https://stackoverflow.com/questions/30004945/why-couldnt-go-read-a-request-properly. 

The gist of it is that we should probably not assume that failing to read the raw body bytes is a user error. I think if the json unmarshal fails (line 89), we could safely assume the user had an error in their JSON and we could safely return a 400 status code.

Comment (mtrinh11) [reply transport.go]: I've reverted this line to just return the err again

**Review by joe94 — APPROVED** [review]

File: transport_test.go [hunk 2]
```diff
@@ -215,7 +239,16 @@ func TestAddWebhookRequestDecoder(t *testing.T) {
...
 			if tc.ExpectedErr != nil {
-				assert.Equal(tc.ExpectedErr, err)
+				assert.True(errors.Is(err, tc.ExpectedErr),
+					fmt.Errorf("error [%v] doesn't contain error [%v] in its err chain",
+						err, tc.ExpectedErr))
+				if tc.ExpectedStatusCode != 0 {
+					var s kithttp.StatusCoder
+					isCoder := errors.As(err, &s)
+					require.True(isCoder, "error isn't StatusCoder as expected")
+					require.Equal(http.StatusBadRequest, s.StatusCode())
```
Comment (kristinapathak) [inline transport_test.go:249]:
```suggestion
					require.Equal(tc.ExpectedStatusCode, s.StatusCode())
```

Comment (mtrinh11) [reply transport_test.go]: implemented!

**Review by kristinapathak — APPROVED** [review]

**Pull request merged**